### PR TITLE
Fix Google auth

### LINKED ISSUE #2383: Authenticating a user with Google is throwing 500 on the 2nd login
**Informations**
- **Node.js version**: 10.13.0
- **npm version**:  6.4.1
- **Strapi version**: v3.0.0-alpha.14.5
- **Database**: mongodb 3.6.5
- **Operating system**: Win10

**What is the current behavior?**
After the first successful authentication with google, additional attempts to authenticate, at least within the next minutes, fail at redirecting to GET /auth/google/callback as long as the user remains signed in with google in the browser. 
Response is 500 at http://localhost:4200/auth/callback/google?error%5Berror%5D=invalid_grant&error%5Berror_description%5D=Malformed%20auth%20code.

Signing off from Google fixes this issue and returns jwt and user response after redirecting to GET /auth/google/callback

**Steps to reproduce the problem**
Authenticate with google
Stay signed in with google
Authenticate again with google

**What is the expected behavior?**
After the first authentication subsequent authentication attempts should also issue a new token.

**Suggested solutions**
I'm not quite sure why this happens since the initial redirect from accounts.google.com/o/oauth2/auth to /connect/google/callback looks pretty much the same on the first and subsequent calls, only difference i could recognize was that the first redirect is encoded while the next ones are not

First attempt, working:
http://localhost:1337/connect/google/callback?code=4%2FogD_bJ10kjNs7l8gqTF25hLsYQPUU-rkatS6jK5shBXcw-lLN0wlJpbSDbyx8zFP2yuDyVLDA1ScgSxLaZxxxxx&scope=openid+email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fplus.me

Next attempt, failing: 
http://localhost:1337/connect/google/callback?code=4/ogCWJEaBayzfL2PoY4NJ3rL-11Wu4FNjrPOxRlFG003qPuGCy7nV6qCnlX1vvh_Dt1cUmJZrFn5ESxxxxx&scope=openid+email+https://www.googleapis.com/auth/plus.me+https://www.googleapis.com/auth/userinfo.email

### LINKED ISSUE #2660: Changes for Google auth provider
- [x] **I have created my request on the Product Board before I submitted this issue**
- [x] **I have looked at all the other requests on the Product Board before I submitted this issue**

**Please describe your feature request:**

Google is going to [shut down the consumer version of Google+](https://www.blog.google/technology/safety-security/project-strobe/) over the next several months and that means that Google provider will stop working as it relies on Google+ as I can see. Are there any plans to migrate existing Google provider to use a recommended [Google Sign-In API](https://developers.google.com/identity/sign-in/web/sign-in#before_you_begin)? Existing apps using Google provider might be impacted once Google+ is taken down. Or do I miss anything and existing provider won't require any changes given the announcement?

### PR DESCRIPTION
**Pull request by lauriejim**
#### Description of what you did:

Since **Google +** was shut down, we had an issue withe the Google provider authentification.
Instead of using by the default **Purest** Google provider (that us Google +), here we use Google API via a custom config of Purest.

As there is no travis tests, if you want to test manually this PR, you will have to follow this documentation.
📚 [Google provider configuration](https://github.com/strapi/strapi-examples/blob/master/login-react/doc/google_setup.md)
And use the [login react](https://github.com/strapi/strapi-examples/tree/master/login-react) app to test it.

This PR close #2660 and close #2383

#### My PR is a:

- [ ] 💥 Breaking change
- [x] 🐛 Bug fix
- [ ] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [ ] Documentation
- [ ] Framework
- [x] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [x] SQLite

### CODE REVIEW

**Comment by LucasMatuszewski** [comment]
Thanks a lot for this fix @lauriejim
In my case it don't want to work. On a server I get this error:
`[2019-07-13T04:17:04.350Z] debug GET \connect\google\callback?code=4\xxxxxxxxxxxxxxxxxxxxxxxxx-x0mxxxxxxxxxxxxxxxxxxxxxxxxxxxxxQ7CuvW8&scope=email+htt ps:\www.googleapis.com\auth\userinfo.email+openid&authuser=0&session_state=3798722a154c87da4728e8e6819657d32259d640..a90c&prompt=none (223 ms)
[2019-07-13T04:17:06.893Z] error TypeError: Cannot read property 'Symbol(Symbol.iterator)' of object                                                                             at callback (D:\Dev\Projects\Edukey_Next_Strapi\strapi\node_modules\strapi-plugin-users-permissions\controllers\Auth.js:142:23)
[2019-07-13T04:17:06.895Z] debug GET \auth\google\callback?error%5Berror%5D=invalid_grant&error%5Berror_description%5D=Malformed%20auth%20code. (79 ms)`

Request goes here:
`Request URL: http://localhost:1337/auth/google/callback?error%5Berror%5D=invalid_grant&error%5Berror_description%5D=Malformed%20auth%20code.`

I'v copied your code manually to this file:
> strapi\node_modules\strapi-plugin-users-permissions\services\Providers.js

my versions:
`"strapi": "^3.0.0-next.17",
    "strapi-admin": "^3.0.0-next.17",
    "strapi-hook-bookshelf": "3.0.0-beta.6",
    "strapi-hook-knex": "3.0.0-beta.6",
    "strapi-plugin-content-manager": "3.0.0-beta.6",
    "strapi-plugin-content-type-builder": "3.0.0-beta.6",
    "strapi-plugin-documentation": "^3.0.0-beta.6",
    "strapi-plugin-email": "3.0.0-beta.6",
    "strapi-plugin-graphql": "^3.0.0-next.11",
    "strapi-plugin-settings-manager": "3.0.0-beta.6",
    "strapi-plugin-upload": "3.0.0-beta.6",
    "strapi-plugin-users-permissions": "^3.0.0-next.17",
    "strapi-utils": "3.0.0-beta.6"`

I'm on Windows 10 local machine, with Node v10.14.1 and NPM 6.4.1.

Do I have to change Setting on Google APIs (I still use Google+ API there).

10 days ago I used @JulioC quickfix and It was working, until I have updated a Strapi version:
https://github.com/strapi/strapi/issues/2383#issuecomment-507877561 (code changed and I don't know

File: packages/strapi-plugin-users-permissions/services/Providers.js [hunk 1]
```diff
@@ -159,17 +159,37 @@ const getProfile = async (provider, query, callback) => {
...
+            "oauth/[version]/{endpoint}": {
+              "__path": {
+                "alias": "oauth",
+                "version": "v3"
+              }
+            }
+          }
+        }
+      }
+      var google = new Purest({provider: 'google', config})
```
Comment (LucasMatuszewski) [inline packages/strapi-plugin-users-permissions/services/Providers.js]: Why `var` ? :)

**Comment by LouisAugry** [comment]
Waiting for this fix ! 😉

**Comment by maxbaluev** [comment]
need it

**Comment by lauriejim** [comment]
I fix the typo.

@LucasMatuszewski you are using a `next` version, I suggest you to use a `beta` version of strapi.
And I re-tested it and it works well. Can you make sure, your Google credentials are correct.
By following my PR description message.

**Comment by LucasMatuszewski** [comment]
@lauriejim thanks for response.

I assume next is unstable future version, under development/tests? 
So in my apps I should use beta version. Next is for contributors?

I just used `npm install` on my project after switching to other computer. So I should specify Starpi version on installing?

**Comment by derrickmehaffy** [comment]
@LucasMatuszewski `next` is used for testing WIP code, absolutely not for any real use. Very good chance stuff is half done, not working, ect there.

Beta is the public use one, you will have to be careful with `npm audit fix` as it will sometimes inject next versions instead of beta ones.

**Comment by LucasMatuszewski** [comment]
Indeed, I used `npm audit fix`. Thanks a lot for clarification.

**Comment by lauriejim** [comment]
Thank you for your feedback @LucasMatuszewski

**Pull request merged**